src/compiler/jsgen.js

```diff
@@ -185,9 +185,9 @@ class JSGenerator {
                 return `(+${this.descendInput(node.target.toType(InputType.BOOLEAN))})`;
             }
             if (node.target.isAlwaysType(InputType.NUMBER_OR_NAN)) {
-                return `(${this.descendInput(node.target)} || 0)`;
+                return `toNotNaN(${this.descendInput(node.target)})`;
```

Comment:
Could add a check to see if the node can be negative zero, and if it can't, maybe use the old `|| 0` method if it's faster. Doubt it'll make a huge difference tho

Reply:
I think I tried this and it only affected like 3 places in the entire test suite, most of those being in the generate comparison testing matrix which isn't very real-world. Think Linux in Scratch had no changes in generated code at all. Don't think it's worth checking that at this time